src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3735,6 +3772,65 @@ class CgroupUtils(object):
                 raise
         return self.assigned_resources is not None
 
+    def add_jobid_to_cgroup_jobs(self, jobid):
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'Adding jobid %s to cgroup_jobs' % jobid)
+        try:
+            with open(self.cgroup_jobs_file, 'a') as f:
+                f.write(jobid + ' ')
```

Comment:
I really like they way you read in the job list in remove_jobid_from_cgroup_jobs(), and I think it would make sense to do the same thing here. It's not that expensive since you already seek to the end of the file when you open it.

Alternatively, if f.tell indicates your offset is zero then you know it's the first entry. Then you only need to append jobid (and no space). For the other cases you would append (' ' + jobid).

What if the cgroup_jobs_file doesn't exist? Do you need to open as 'a+'?

Reply:
I don't understand. This function only needs to append one jobid so there is no reading involved. 

If the `cgroup_jobs` file does not exist, `open()` will create it. I was wrong about the `+` mode being necessary. I have fixed other places.

Reply:
In the current implementation, there will always be a space following the last jobid in the file. I don't think that will affect the behavior, but it just feels a little sloppy. You could do something like...
'''
with open(self.cgroup_jobs_file, 'a') as f:
    if f.tell() > 0:
        f.write(' ')
    f.write(jobid)
'''

Reply:
Fixed. Thanks.